README.md

```diff
@@ -130,61 +130,77 @@ $ source deps/zephyr/zephyr-env.sh
+this one it's not really needed.
+
 Then connect the Arduino 101 to your host with a USB A/B cable. Press the
 Master Reset button on the Arduino 101 and after a few seconds type:
 
 ```bash
-$ make dfu-arc
+$ make dfu-all
 ```
 
-This will flash the ARC image to the device using the dfu-util program.
-(Note, this is not running the JavaScript engine and application yet, that will
-occur on the x86 side below.)
+This will flash both the images to the device using the dfu-util program.
 
 If you get a permission error, make sure you followed the 'Join the plugdev
 group' instructions above for this user. You shoudn't need to run this command
 with `sudo`.
 
-(The ARC image doesn't change very often so you won't need to do this again
-unless you pull updates to the source tree, and usually not even then.)
+After this flashing completes successfully, reboot the device with the Master
+Reset button to start the application. After a few seconds the onboard LEDs
+should start cycling.
 
-#### x86 application image
-Next, build the x86 image which includes the JerryScript engine and the
-ZJS runtime support, along with your JavaScript application:
+You have built and run your first ZJS application!
 
-```bash
-$ make JS=samples/TrafficLight.js
-```
+If you want to make changes to the application, or run a different .js sample,
+you just need to repeat the steps the desired JavaScript filename.
 
-The JS= argument lets you provide the path to your application. The TrafficLight
-sample is a good first choice because you don't need to wire up any additional
-hardware. It just blinks onboard LEDs on your Arduino 101. Also, for many of
-the samples you will want to hook up the serial console (see below), but for
-this one it's not really needed.
+#### x86 application image only
+If you want to build the x86 and ARC images separately, or if you just want to
+build another .js sample that uses the same sub-modules, then you can  build x86
+only with
 
-Finally, you're ready to install and run your application. If you just flashed
-the ARC image as above, your Arduino 101 will already be waiting for more DFU
-commands. If not, press the Master Reset button, and either way do:
+```bash
+$ make JS=samples/xxx.js
+```
 
 ```bash
 $ make dfu
 ```
 
-After this flashing completes successfully, reboot the device with the Master
-Reset button to start the application. After a few seconds the onboard LEDs
-should start cycling.
+#### ARC support image only
+To build and flash the ARC support image only
 
-You have built and run your first ZJS application!
+```bash
+$ make JS=samples/xxx.js arc
+```
 
-If you want to make changes to the application, or run a different .js sample,
-you just need to repeat the last two steps with the desired JavaScript filename.
+```bash
+$ make dfu-arc
+```
+
+If you run a different .js sample that uses different sub-modules, for example,
+from AIO.js to I2C.js, then you'll need to repeat the last two steps to flash
+both images.
```

Comment:
What happens if they forget? I guess it would say unknown message type? That's probably good enough, but you could paste that error message here and explain how to correct it.